datasets/interpress_news_category_tr/interpress_news_category_tr.py

```diff
@@ -0,0 +1,114 @@
+    """BuilderConfig for InterpressNewsCategoryTR."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for InterpressNewsCategoryTR.
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(InterpressNewsCategoryTRConfig, self).__init__(**kwargs)
+
+
+class InterpressNewsCategoryTR(datasets.GeneratorBasedBuilder):
+    """Turkish News Category Dataset (270K) - Interpress Media Monitoring Company"""
+
+    BUILDER_CONFIGS = [
+        InterpressNewsCategoryTRConfig(
+            name="interpress_news_category_tr_270k",
+            version=datasets.Version("1.0.0"),
+            description="Turkish News Category Dataset (270K) - Interpress Media Monitoring Company",
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "category": datasets.features.ClassLabel(
+                        names=[
+                            "aktuel",
+                            "bilisim",
+                            "egitim",
+                            "ekonomi",
+                            "gida",
+                            "iletisim",
+                            "kultursanat",
+                            "magazin",
+                            "saglik",
+                            "savunma",
+                            "seyahat",
+                            "siyasi",
+                            "spor",
+                            "teknoloji",
+                            "ticaret",
+                            "turizm",
+                            "yasam",
+                        ]
+                    ),
+                    "id": datasets.Value("int32"),
+                    "title": datasets.Value("string"),
+                    "publishdatetime": datasets.Value("string"),
+                    "content": datasets.Value("string"),
+                    "categorycode": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        dl_dir = dl_manager.download_and_extract(_DOWNLOAD_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN, gen_kwargs={"filepath": os.path.join(dl_dir, _DATASET_URLS["train"])}
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST, gen_kwargs={"filepath": os.path.join(dl_dir, _DATASET_URLS["test"])}
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """Generate InterpressNewsCategoryTR examples."""
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            reader = csv.DictReader(f, delimiter="\t", quoting=csv.QUOTE_NONE)
+            csv.field_size_limit()
```

Comment:
What does this line of code do ?

Reply:
Since the number of lines in the TSV file is large, it was necessary to define it as "csv.field_size_limit (sys.maxsize)" in our local. However, we are removing the "csv.field_size_limit ()" line since it is not needed for now.